### PR TITLE
build: update to typescript v3.8

### PR DESCRIPTION
**Pull request by devversion**
Updates to TypeScript v3.8. To ensure that Angular Components is still
compatible with older supported TypeScript versions, a new integration
test has been introduced.

Unlike in `angular/angular`, our packages have a _lot_ more entry-points, so manually
creating files that import all entry-points for the compatibility test is not suitable. This
required a more dynamic and less verbose solution.

Also, audited all TypeScript AST work to check for the new AST changes from v3.8. It looks like there is no action needed, but the `class-name` `ng-update` rule seems to need a general cleanup to handle namespaced exports/imports better. Added a comment w/ JIRA ticket.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -484,6 +484,23 @@ jobs:
       # Run project tests with NGC and View Engine.
       - run: bazel test src/... --build_tag_filters=-docs-package,-e2e --test_tag_filters=-e2e --config=view-engine --build_tests_only
 
+  # ----------------------------------------------------------------------------
+  # Job that runs all Bazel integration tests.
+  # ----------------------------------------------------------------------------
+  integration_tests:
+    <<: *job_defaults
+    resource_class: xlarge
```
Comment (devversion) [inline .circleci/config.yml]: I think we should do some audits on resource classes in the future. Currently we have them set in all expensive jobs (such as building npm packages). Should I create a JIRA for tracking? I think there might be room for reductions.

Comment (josephperrott) [reply .circleci/config.yml]: Yes, you should create a JIRA in dev-infra, honestly we need to come up with a process for this type of auditing, or even an automation ideally.

Comment (devversion) [reply .circleci/config.yml]: Done.

File: integration/ts-compat/import-all-entry-points.bzl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    import * as {0} from "@angular/{1}/{2}";
+    export {{ {0} }};
+  """.format(identifier, pkg_name, entry_point)
+
+"""
+  Creates a file that imports all entry-points as namespace. The namespaces will be
+  re-exported. This ensures that all entry-points can successfully compile.
+"""
+
+def generate_import_all_entry_points_file():
```
Comment (jelbourn) [inline integration/ts-compat/import-all-entry-points.bzl]: Is there a way we could consolidate on a single, canonical list of packages for the repo? Right now I think there's a handful of places that need to be updated when we add a new package. Either that, or just documenting all the places that need to be updated

Comment (devversion) [reply integration/ts-compat/import-all-entry-points.bzl]: 💯 Definitely agreed. I've been trying to reduce this type of duplication (as seen with entry-points in the system config etc.). 

I'll work on a follow-up that gives us a better list of release packages. The problem always comes from the fact that we can't easily access these variables outside of Bazel (e.g. in the release scripts).

**Pull request merged**